src/reducers/entities/channels.js

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) 2016 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import {ChannelTypes, UsersTypes} from 'constants';
+import {combineReducers} from 'redux';
+
+function currentId(state = '', action) {
+    switch (action.type) {
+    case UsersTypes.LOGOUT_SUCCESS:
+        return '';
+    default:
+        return state;
+    }
+}
+
+function channels(state = {}, action) {
+    const nextState = {...state};
+    switch (action.type) {
+    case ChannelTypes.RECEIVED_CHANNEL:
+        return {
+            ...state,
+            [action.channel.id]: action.channel
+        };
+
+    case ChannelTypes.RECEIVED_CHANNELS:
+        for (const channel of action.channels) {
+            nextState[channel.id] = channel;
+        }
+        return nextState;
+
+    case ChannelTypes.LEAVE_CHANNEL:
+    case ChannelTypes.RECEIVED_CHANNEL_DELETED:
+        Reflect.deleteProperty(nextState, action.channel_id);
+        return nextState;
+
+    case UsersTypes.LOGOUT_SUCCESS:
+        return {};
+
+    default:
+        return state;
+    }
+}
+
+function myMembers(state = {}, action) {
+    const nextState = {...state};
+    let member;
+    switch (action.type) {
+    case ChannelTypes.RECEIVED_MY_CHANNEL_MEMBER:
+        const channelMember = action.channelMember;
+        return {
+            ...state,
+            [channelMember.channel_id]: channelMember
+        };
+
+    case ChannelTypes.RECEIVED_MY_CHANNEL_MEMBERS:
+        for (const cm of action.channelMembers) {
+            nextState[cm.channel_id] = cm;
+        }
+        return nextState;
+
+    case ChannelTypes.RECEIVED_CHANNEL_PROPS:
+        member = {...state[action.channel_id]};
```

Comment:
Are you sure this copies the correct thing? I don't know whether this is treated as `(...state)[action.channel_id` or `...(state[action.channel_id])`